File: KiddyTube/app/src/main/java/ae/kiddytube/app/parent/ParentSession.kt
package ae.kiddytube.app.parent

/**
 * In-memory parent unlock session. Survives activity recreation within the process
 * but not process death — Recents cold-resume still requires PIN after expiry.
 */
object ParentSession {
    private const val UNLOCK_TTL_MS = 5 * 60 * 1000L
    @Volatile
    private var unlockedUntilMs: Long = 0L

    fun grant(nowMs: Long = System.currentTimeMillis()) {
        unlockedUntilMs = nowMs + UNLOCK_TTL_MS
    }

    fun clear() {
        unlockedUntilMs = 0L
    }

    fun isActive(nowMs: Long = System.currentTimeMillis()): Boolean =
        nowMs < unlockedUntilMs
}

File: kiddytube_flutter/lib/parent/parent_session.dart
/// In-memory parent unlock session (~5 minutes), mirrors Kotlin [ParentSession].
/// Activity calls [touch] to slide the TTL while the parent settings screen is open.
class ParentSession {
  ParentSession._();

  static const unlockTtlMs = 5 * 60 * 1000;
  static int _unlockedUntilMs = 0;

  static void grant([int? nowMs]) {
    final now = nowMs ?? DateTime.now().millisecondsSinceEpoch;
    _unlockedUntilMs = now + unlockTtlMs;
  }

  /// Extends the session when still active (sliding expiration).
  static void touch([int? nowMs]) {
    if (!isActive(nowMs)) return;
    grant(nowMs);
  }

  static void clear() {
    _unlockedUntilMs = 0;
  }

  static bool isActive([int? nowMs]) {
    final now = nowMs ?? DateTime.now().millisecondsSinceEpoch;
    return now < _unlockedUntilMs;
  }

  /// Milliseconds remaining, or 0 if expired.
  static int remainingMs([int? nowMs]) {
    final now = nowMs ?? DateTime.now().millisecondsSinceEpoch;
    final left = _unlockedUntilMs - now;
    return left > 0 ? left : 0;
  }
}
